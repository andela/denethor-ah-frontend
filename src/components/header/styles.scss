//colors
$white: #ffffff;
$black: #000000;
$gray: #818181;

.header{
  background: $white;
  border-bottom: 0.1em solid $gray;
  display: flex;
  justify-content: space-between;
  padding: 0.8em;
  > button {
    background: inherit;
    display: block;
    width: 40%;
    border: none;
  }
}

.logo-container {
  margin-left: 5%;
  display: flex;
  align-items: center;

  .logo-text {
    h3 {
      color: black;
      font-family: 'SairaCondensed-Regular';
      margin-left: 0.4em;
      font-size: 1.6em;
      letter-spacing: 0.16rem;
    }
  }
}

.nav-container {
  margin-right: 5%;
  display: flex;
  align-items: center;

  ul {
    display: flex;
    align-items: center;

    li {
        display: flex;
        justify-content: space-around;
        margin-left: 0.8em;
      }
  }

  button.signup-link, button.login-link {
    background: inherit;
    border: none;
    text-decoration: none;
    color: $gray;
    font-family: 'Montserrat-SemiBold';
    font-size: 1.2em;
  }
}

  .mobile-header {
    display: none
  }

  @media all and (max-width:550px) and (min-width:0px) {
    .header > button, .nav-container {
      display: none;
    }
    
    .mobile-header {
      display: flex;
      justify-content: space-between;
      align-items: center;
      width: 100%;

      > a {
        color: black;
        text-decoration: none;
      }

      .mobile-header-text {
        font-family: 'SairaCondensed-Regular';
        
        h3 {
          margin: 0.5rem 0;
          font-size: 1.3em;
        }
      }
    }
  }