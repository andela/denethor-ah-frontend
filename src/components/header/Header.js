import React from 'react';
import logo from '../../../public/assets/img/ah-logo.svg';
import { Link } from 'react-router-dom';
import searchIcon from '../../../public/assets/img/search-icon.svg';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import './styles.scss';

library.add(faBars);

const Header = () => {
  return (
    <div className="header">
      <div className="logo-container">
        <div className="logo-image"> 
          <img src={logo} alt="Author's haven logo"/>
        </div>
        <div className="logo-text">
          <h3>AUTHOR&apos;S HAVEN</h3>
        </div>
      </div>
      <div className="nav-container">
        <ul>
          <li><Link to='/'><img src={searchIcon} alt="search icon"/></Link></li>
          <li><Link to='/' className="signup-link">Signup</Link></li>
          <li><Link to='/' className="login-link">Login</Link></li>
        </ul>
      </div>
      <div className="mobile-header"> 
        <FontAwesomeIcon icon="bars" color="#818181" size="2x"/>
        <div className="mobile-header-text"><h3>AUTHOR&apos;S HAVEN</h3></div>
        <div className="mobile-header-search-icon"><img src={searchIcon} alt="Mobile view search icon"/></div>
      </div>
    </div>
  )
}

export default Header;