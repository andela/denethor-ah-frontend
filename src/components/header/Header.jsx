import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars, faSearch } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import AuthButton from './authButton';
import Widgets from './widgets';
import axios from '../../utils/axiosConfig';
import './styles.scss';
import { logout } from '../../redux/actions/auth';
import { removeOwnProfile } from '../../redux/actions/profile';
import { toggleMobileSideBar } from '../../redux/actions/elements';
import MobileSideBar from '../dashboard/sideBar/MobileSideBar';
import SearchWidget from './widgets/searchWidget/SearchWidget';

library.add(faBars);
library.add(faSearch);

export const Header = ({ history, dispatch, isLoggedIn, toggleMobileSideBar, sideBarActive }) => {
  const handleNavSignup = () => {
    history.push('/signup');
  };

  const handleNavLogin = () => {
    if(history.location.pathname === '/') {
      history.push('/login');
    }

  };

  const handleNavLogout = async () => {
    await axios.get(`${process.env.API_ROOT_URL}/users/logout`);
    localStorage.clear();
    dispatch(logout());
    dispatch(removeOwnProfile());
    history.push('/');
  };

  const handleLogoClick = () => {
    history.push('/');
  };

  const theme = 'theme__black';

  let fontColors = {
    theme__black: '#ffffff'
  };

  const iconColor = fontColors[theme];

  return (
    <div className={`header ${theme}`}>
      <button onClick={handleLogoClick} className='logo-container show-for-large'>
        <div className="logo-image">
          <img src='/assets/img/ah-logo.svg' alt="Author's haven logo" />
        </div>
        <div className="logo-text">
          <h3>AUTHOR&apos;S HAVEN</h3>
        </div>
      </button>
      <div className="nav-container show-for-large">
        <ul>
          <li><SearchWidget /></li>
          {
            isLoggedIn
            ?
              <li><Widgets /></li>
            :
            <React.Fragment>
              <li><AuthButton type="Signup" /></li>
              <li><AuthButton type="Login" /></li>
            </React.Fragment>
          }
        </ul>
      </div>
      <div className="mobile-header hide-for-large">
        <FontAwesomeIcon icon="bars" color={iconColor || '#818181'} size="2x" onClick={toggleMobileSideBar} />
        <Link to='/'>
          <div className="mobile-header-text"><h3>AUTHOR&apos;S HAVEN</h3></div>
        </Link>
        <SearchWidget />
      </div>

      <MobileSideBar
        isLoggedIn={isLoggedIn}
        handleNavLogout={handleNavLogout}
        handleNavSignup={handleNavSignup}
        handleNavLogin={handleNavLogin}
        sideBarActive={sideBarActive}
        toggleMobileSideBar={toggleMobileSideBar}
      />
    </div>
  )
};

Header.propTypes = {
  dispatch: PropTypes.func,
  history: PropTypes.object,
  isLoggedIn: PropTypes.bool,
  toggleMobileSideBar: PropTypes.func,
  sideBarActive: PropTypes.bool
}

const mapStateToProps = ({ auth: { isLoggedIn = false }, elementStatuses: { sideBarActive = false } }) => 
  ({ isLoggedIn, sideBarActive });

const mapDispatchToProps = (dispatch) => ({
  toggleMobileSideBar: () => dispatch(toggleMobileSideBar()),
  dispatch
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Header));
