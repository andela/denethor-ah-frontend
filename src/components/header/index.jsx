import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBars } from '@fortawesome/free-solid-svg-icons';
import logo from '../../images/ah-logo.svg';
import searchIcon from '../../images/search-icon.svg';
import './styles.scss';

library.add(faBars);

export const Header = ({ history }) => {
  const handleNavSignup = () => {
      history.push('/signup');
  };

  const handleNavLogin = () => {
      history.push('/login');
  };

  const handleLogoClick = () => {
    history.push('/');
  };

  return (
    <div className="header">
      <button onClick={handleLogoClick}>
        <div className="logo-container">
          <div className="logo-image"> 
            <img src={`/${logo}`} alt="Author's haven logo"/>
          </div>
          <div className="logo-text">
            <h3>AUTHOR&apos;S HAVEN</h3>
          </div>
        </div>
      </button>
      <div className="nav-container">
        <ul>
          <li><Link to='/'><img src={`/${searchIcon}`} alt="search icon"/></Link></li>
          <li><button onClick={handleNavSignup} className="signup-link">Signup</button></li>
          <li><button onClick={handleNavLogin} className="login-link">Login</button></li>
        </ul>
      </div>
      <div className="mobile-header"> 
      <FontAwesomeIcon icon="bars" color="#818181" size="2x"/>
        <Link to='/'>
          <div className="mobile-header-text"><h3>AUTHOR&apos;S HAVEN</h3></div>
        </Link>
        <div className="mobile-header-search-icon"><img src={`/${searchIcon}`} alt="Mobile view search icon"/></div>
      </div>
    </div>
  )
}

export default withRouter(Header);
