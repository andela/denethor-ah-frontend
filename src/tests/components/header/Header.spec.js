import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../../components/header/Header';

test("it renders correctly as a static component", () => {
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot();
});

test("it calls login render function on click of login button", () => {
  const history = { location: { pathname: '/' } };
  const handleLogin = jest.fn();
  const handleSignup = jest.fn();
  const loadStats = jest.fn();
  const props = { history, handleLogin, handleSignup, loadStats };
  const wrapper = shallow(<Header {...props}/>);
  wrapper.find('button').at(1).simulate('click');
  expect(handleSignup).toHaveBeenCalled();
});

test("it calls login render function on click of signup button", () => {
  const history = { location: { pathname: '/' } };
  const handleLogin = jest.fn();
  const handleSignup = jest.fn();
  const loadStats = jest.fn();
  const props = { history, handleLogin, handleSignup, loadStats };
  const wrapper = shallow(<Header {...props}/>);
  wrapper.find('button').at(2).simulate('click');
  expect(handleLogin).toHaveBeenCalled();
});

test("it calls loadStats function on click of logo", () => {
  const history = { location: { pathname: '/' } };
  const handleLogin = jest.fn();
  const handleSignup = jest.fn();
  const loadStats = jest.fn();
  const props = { history, handleLogin, handleSignup, loadStats };
  const wrapper = shallow(<Header {...props}/>);
  wrapper.find('button').at(0).simulate('click');
  expect(loadStats).toHaveBeenCalled();
});
