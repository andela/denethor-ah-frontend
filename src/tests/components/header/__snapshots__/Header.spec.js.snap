// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly as a static component 1`] = `
<div
  className="header theme__black"
>
  <button
    className="logo-container show-for-large"
    onClick={[Function]}
  >
    <div
      className="logo-image"
    >
      <img
        alt="Author's haven logo"
        src="/assets/img/ah-logo.svg"
      />
    </div>
    <div
      className="logo-text"
    >
      <h3>
        AUTHOR'S HAVEN
      </h3>
    </div>
  </button>
  <div
    className="nav-container show-for-large"
  >
    <ul>
      <li>
        <Connect(withRouter(SearchWidget)) />
      </li>
      <li>
        <Connect(withRouter(AuthButton))
          type="Signup"
        />
      </li>
      <li>
        <Connect(withRouter(AuthButton))
          type="Login"
        />
      </li>
    </ul>
  </div>
  <div
    className="mobile-header hide-for-large"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#ffffff"
      fixedWidth={false}
      flip={null}
      icon="bars"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <Link
      replace={false}
      to="/"
    >
      <div
        className="mobile-header-text"
      >
        <h3>
          AUTHOR'S HAVEN
        </h3>
      </div>
    </Link>
    <Connect(withRouter(SearchWidget)) />
  </div>
  <MobileSideBar
    handleNavLogin={[Function]}
    handleNavLogout={[Function]}
    handleNavSignup={[Function]}
  />
</div>
`;
